pkgs/tools/misc/mstflint/default.nix

```diff
@@ -4,23 +4,84 @@
 , libibmad
 , openssl
 , zlib
+, xz
+, expat
+, boost
+, curl
+, pkg-config
+, autoreconfHook
+, libxml2
+, pciutils
+, busybox
+, python3
+, minimalBuild ? false
 }:
 
 stdenv.mkDerivation rec {
   pname = "mstflint";
-  version = "4.17.0-1";
+  version = "4.25.0-1";
 
   src = fetchurl {
     url = "https://github.com/Mellanox/mstflint/releases/download/v${version}/mstflint-${version}.tar.gz";
-    sha256 = "030vpiv44sxmjf0dng91ziq1cggwj33yp0l4xc6cdhnrv2prjs7y";
+    hash = "sha256-nYGiWfr8a3q3+bGUb1ovLrAS8/LnEJf+4inIEllW95s=";
   };
 
+  nativeBuildInputs = [
+    autoreconfHook
+    pkg-config
+  ];
+
   buildInputs = [
     libibmad
-    openssl
     zlib
+    libxml2
+    openssl
+  ] ++ lib.optional (!minimalBuild) [
+    boost.dev
+    curl.dev
+    expat.dev
+    xz.dev
+    python3
+  ];
+
+  # tracked upstream at
+  #  - https://github.com/Mellanox/mstflint/pull/855
+  #  - https://github.com/Mellanox/mstflint/issues/856
+  # remove if merged
+  patches = [
+    ./0001-fix-dynamic-linking.patch
+    ./0002-fix-mstprivhost.py.patch
+    ./0003-fix-mtcr.py.patch
   ];
 
+  preConfigure = ''
+    export CPPFLAGS="-I$(pwd)/tools_layouts -isystem ${libxml2.dev}/include/libxml2"
+    export INSTALL_BASEDIR=$out
+  '';
+
+  # Cannot use wrapProgram since the python script's logic depends on the
+  # filename and will get messed up if the executable is named ".xyz-wrapped".
+  # That is why the python executable and runtime dependencies are injected
+  # this way.
+  prePatch = lib.optionals (!minimalBuild) ''
+    substituteInPlace common/python_wrapper.sh \
+      --replace \
+      'exec $PYTHON_EXEC $SCRIPT_PATH "$@"' \
+      'export PATH=$PATH:${lib.makeBinPath [ (placeholder "out") pciutils busybox]}; exec ${python3}/bin/python3 $SCRIPT_PATH "$@"'
+  '';
+
+  configureFlags = [
+    "--enable-xml2"
+  ] ++ lib.optional (!minimalBuild) [
```

Comment:
Is there really a good reason why anyone would want a minimal build?

Reply:
Probably not everyone, but for more dedicated or appliance use cases the minimal build removes the interpreter dependency and tools for fetching firmware updates from the Internet. One has to know his hardware in advance in this case and provide the correct firmware binary himself.

Reply:
The problem with these flags is that hydra will only ever build the default version. So if there is no harm for everyone to have the full version I would vote against the feature flag.

Reply:
Fair point. I'd like to use the minimal version myself downstream, but can also override there.

Reply:
I renamed the option and added some comment.